### ISSUE TITLE
Error: ‘tgeometry’ in namespace ‘open3d’ does not name a type

### ISSUE THREAD

**Issue by Matnay** [intel-isl/Open3D#2357]
**IMPORTANT: Please use the following template to report the bug.**
_____

**Describe the bug**
error: ‘tgeometry’ in namespace ‘open3d’ does not name a type. The error message is triggered when I try to use it as a function argument as shown below.
```
void open3d_convert(const open3d::tgeometry::PointCloud& pointcloud)
```

**To Reproduce**
Installation:

  - git clone --recursive https://github.com/intel-isl/Open3D
  - cd Open3D && git submodule update --init --recursive
  - source util/install_deps_ubuntu.sh
  - mkdir build && cd build
  - cmake -DBUILD_UNIT_TESTS=ON -DBUILD_EIGEN3=ON -DBUILD_GLEW=ON -DBUILD_GLFW=ON -DBUILD_JSONCPP=ON -DBUILD_PNG=ON -DGLIBCXX_USE_CXX11_ABI=ON -DPYTHON_EXECUTABLE=/usr/bin/python ..
  - make -j8
  - sudo make install 
Upon compiling the code the build fails.

**Expected behavior**
The code should compile and work on Linux. 

**Environment:**

- Operating system: Ubuntu 18.04.5 LTS
- Python version:  Python 2.7
- Open3D version: 0.10.1
- Is this remote workstation?:  no
- How did you install Open3D?:  build from source
- Compiler version (if built from source):  gcc 7.5

**Additional context**
Adding the necessary headers to Open3D.h.in solves the issue and I have created a PR for the same. #2137

**Comment by nkhedekar** [comment]
Having the same issue, #2137 solves it

**Comment by theNded** [comment]
Fixed in #2137, closed for now
